### PR TITLE
Extract Link and move Eth to evm folder

### PR DESCRIPTION
**Pull request by dimriou**

### CODE REVIEW

File: core/chains/evm/assets/currencies.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
+package assets
+
+import (
```
Comment (dhaidashenko) [inline core/chains/evm/assets/currencies.go:3]: nit: should we rename this file to eth? It only contains eth implementation right now. Also this file is large enough to discourage others from adding more code here that is not directly related to eth

**Review by dhaidashenko — APPROVED** [review]
LGTM

**Review by jmank88 — APPROVED** [review]

File: core/chains/evm/config/config.go [hunk 2]
```diff
@@ -6,7 +6,9 @@ import (
 
 	gethcommon "github.com/ethereum/go-ethereum/common"
 
-	"github.com/smartcontractkit/chainlink/v2/core/assets"
+	relayassets "github.com/smartcontractkit/chainlink-relay/pkg/assets"
```
Comment (prashantkumar1982) [inline core/chains/evm/config/config.go:9]: relayassets seems ok for now.
But will be a headache once we rename chainlink-relay to chainlink-common

**Review by prashantkumar1982 — APPROVED** [review]

File: core/chains/evm/config/config.go (hunk 2 above)
Comment (dimriou) [reply core/chains/evm/config/config.go:9]: Agreed, but once we rename the repo, we would have to change the import path as well so let's do a cleanup then.

**Review by jmank88 — APPROVED** [review]

**Review by krehermann — APPROVED** [review]
